newm-server/src/main/kotlin/io/newm/server/features/user/repo/UserRepositoryImpl.kt

```diff
@@ -35,13 +37,15 @@ internal class UserRepositoryImpl(
     private val linkedInUserProvider: LinkedInUserProvider,
     private val appleUserProvider: AppleUserProvider,
     private val twoFactorAuthRepository: TwoFactorAuthRepository,
+    private val configRepository: ConfigRepository,
 ) : UserRepository {
 
     private val logger: Logger by inject { parametersOf(javaClass.simpleName) }
 
     override suspend fun add(user: User): UUID {
         logger.debug { "add: user = $user" }
 
+        user.checkWhitelist()
```

Comment:
@AndrewWestberg We should also add this check at `findOrAdd()` to prevent unauthorized users to join via social.

Reply:
Good catch. Added.